packages/ra-core/src/auth/useLogoutIfAccessDenied.ts

```diff
@@ -46,76 +46,76 @@ const useLogoutIfAccessDenied = (): LogoutIfAccessDenied => {
     const logoutIfAccessDenied = useCallback(
         (error?: any) =>
             authProvider
-                .checkError(error)
-                .then(() => false)
-                .catch(async e => {
-                    const logoutUser = e?.logoutUser ?? true;
-                    //manual debounce
-                    if (timer) {
-                        // side effects already triggered in this tick, exit
-                        return true;
-                    }
-                    timer = setTimeout(() => {
-                        timer = undefined;
-                    }, 0);
+                ? authProvider
```

Comment:
I think this method is a bit too long for a legible ternary. I'd prefer a *leave early*.

Reply:
changed